crates/pathfinder/src/sync/stream.rs

```diff
@@ -49,6 +49,54 @@ impl<T: Send + 'static> SyncReceiver<T> {
         SyncReceiver::from_receiver(rx)
     }
 
+    /// Adds a stage which chunks the incoming elements into a vector before
+    /// passing it on.
+    ///
+    /// `capacity` specifies the number of elements, `buffer` specifies the
+    /// output buffering.
+    pub fn try_chunks(mut self, capacity: usize, buffer: usize) -> SyncReceiver<Vec<T>> {
+        let (tx, rx) = tokio::sync::mpsc::channel(buffer);
+
+        std::thread::spawn(move || {
+            let mut chunk = Vec::with_capacity(capacity);
+            let mut peer = PeerId::random();
+
+            while let Some(input) = self.inner.blocking_recv() {
```

Comment:
We should also handle a scenario where the inner rx channel is being closed but the vector is still partially filled which means it should be sent to the next stage to avoid losing those items.

Reply:
You know. I had that in my mind when I first started and then just completely lost it.. thanks!